### PR TITLE
ci(PPDSC-2017): newskit integration test

### PR DESCRIPTION
**Pull request by Vanals**
PPDSC-2017

**What**

- Created new CircleCi job `test_newskit_in_nextjs_app`.
It Installs needed packages (yalc, create-next-app, etc.)
Creates the NextJs app
Prepares the NExt App adding files(for cy) and overriding the index.js, also installs other packages.
Once the app is ready, through `start-server-and-test` serves the app and tests it using cy.

- Created folder in `scripts` containing few template files for this app.

- Also replicated the issue we released, importing the scenario from the wrong file. The app crashed and the test failed.
it failed even before cy opening the page.  `start-server-and-test` does check already for 200 before testing.

@jps 
Cy might actually not be needed, BUT is a nice setting and nice to have if we foresee adding CY integration tests in the future. If not `start-server-and-test` will be all we need and we can remove CY reducing the job time by around 1 min.

**I have done:**
- [ ] Written unit tests against changes
- [ ] Written functional tests against the component and/or NewsKit site
- [ ] Updated relevant documentation

**I have tested manually:**
- [ ] The feature's functionality is working as expected on Chrome, Firefox, Safari and Edge
- [ ] The screen reader reads and flows through the elements as expected.
- [ ] There are no new errors in the browser console coming from this PR.
- [ ] When visual test is not added, it renders correctly on different browsers and mobile viewports (Safari, Firefox, small mobile viewport, tablet)
- [ ] The Playground feature is working as expected

**Before:**

**After:**

**Who should review this PR:**

**How to test:**

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -860,6 +860,55 @@ jobs:
...
+            cp ./scripts/templates-integration-test/index.txt ./newskit-integration-test/pages/index.js
+            cp ./scripts/templates-integration-test/cypress.json ./newskit-integration-test/
+            mkdir -p ./newskit-integration-test/cypress/integration
+            cp ./scripts/templates-integration-test/theme-checker.spec.js ./newskit-integration-test/cypress/integration
+
+        
+            echo "Installing needed packages into the NextJS app"
+
+            cd newskit-integration-test
+            yalc add newskit
```
[the remark this answers is not in the document]
Comment (Vanals) [reply .circleci/config.yml:894]: Yeah, I was having some issues when using `yarn link`. https://github.com/facebook/react/issues/14257.
Only `yalc` helped me. Not sure if at the end worked cos react needed to be downgraded. Anyway, overall, yalc seem to be better thank yarn link!

**Review by Xin00163 — APPROVED** [review]

File: Makefile [hunk 2]
```diff
@@ -15,6 +15,7 @@ BASE_URI = ${SITE_BASE_URL}${BASE_PATH}/
 install:
 	yarn install --frozen-lockfile
 
+# When changing the version, make sure it marches the one used in the `test_newskit_in_nextjs_app` CircleCI's job.
```
Comment (evgenitsn) [inline Makefile:18]: A tiny typo
```suggestion
# When changing the version, make sure it matches the one used in the `test_newskit_in_nextjs_app` CircleCI's job.
```

**Review by mutebg — APPROVED** [review]
super cool, this is very much how I imagined as well 👏

**Review by evgenitsn — DISMISSED** [review]

**Review by jps — APPROVED** [review]
Looks good!

**Review by evgenitsn — APPROVED** [review]

**Review by mutebg — APPROVED** [review]

**Pull request merged**